python/taichi/lang/_ndarray.py

```diff
@@ -212,7 +213,10 @@ def _pad_key(self, key):
             key = ()
         if not isinstance(key, (tuple, list)):
             key = (key, )
-        assert len(key) == len(self.arr.total_shape())
+        if len(key) != len(self.arr.total_shape()):
+            raise TaichiIndexError(
+                f'{len(self.arr.total_shape())}d ndarray indexed with {len(key)}-d indices: {key}'
```

Comment:
We should use one of `nd` or `n-d` in all places, we should not let both of them appear in the message.